### DISCUSSION TITLE
Missing delegate 'MIFF' in Magick++

### DISCUSSION THREAD

**Discussion by basidale** [ImageMagick/ImageMagick#3355] [Help] [answered]
Hello !

I am working on a C++ project which uses Magick++ so as to save image files in various formats (including .tiff and .dpx). I encountered a problem when i tried to write a .tiff file using the method Image::write. The following exception is trigerred :  

`NoEncodeDelegateForThisImageFormat `MIFF' @ error/constitute.c/WriteImage/1272 .`

I've built ImageMagick on Windows from sources, and i use the MSVC compiler. I noticed that when i select the Debug profile in Visual Studio, the write operation is correctly performed.

Here is the output of the command "convert -list configure" :

`Path: [built-in]

Name                  Value
-------------------------------------------------------------------------------
DELEGATES             bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zip zlib
FEATURES              Debug Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
MAGICK_TEMPORARY_PATH C:/Users/jacqu/AppData/Local/Temp
NAME                  ImageMagick
QuantumDepth          Q16
convert.exe: UnableToOpenConfigureFile `configure.xml' @ warning/configure.c/GetConfigureOptions/702.
 `
Could you help me solve this problem?

**Comment by snibgo** [comment]
```
convert.exe: UnableToOpenConfigureFile configure.xml
```
That is a problem that needs to be solved. Until it is, you will encounter other problems.

What version of IM?

I suspect your installation had a problem. I don't use Visual Studio, and can't advise.

**Comment by basidale** [comment, marked as answer]
I just solved this issue : the problem was that i kept the .dll of the binary distribution of ImageMagick next to the executable, but the .lib files used for linking where those from my build.

Thank you nevertheless !
